XamlControlsGallery/ControlPages/ListViewPage.xaml.cs

```diff
@@ -56,6 +85,242 @@ private void SelectionModeComboBox_SelectionChanged(object sender, SelectionChan
+        {
+            e.AcceptedOperation = DataPackageOperation.Move;
+        }
+
+        private void Source_DragOver(object sender, DragEventArgs e)
+        {
+            e.AcceptedOperation = DataPackageOperation.Move;
+        }
+
+        private async void Target_Drop(object sender, DragEventArgs e)
+        {
+            if (e.DataView.Contains(StandardDataFormats.Text))
+            {
+
+                DragOperationDeferral def = e.GetDeferral();
+                string s = await e.DataView.GetTextAsync();
+                string[] items = s.Split('\n');
+                foreach (string item in items)
+                {
+
+                    // Create Contact object from string, add to existing target ListView
+                    string[] info = item.Split(" ", 3);
+                    Contact temp = new Contact(info[0], info[1], info[2]);
+                    
+                    // Find the insertion index:
+                    Windows.Foundation.Point pos = e.GetPosition(DragDropListView2.ItemsPanelRoot);
+
+                    // Find height of a random item from the target list view
+                    ListViewItem sampleItem = (ListViewItem)DragDropListView.ContainerFromIndex(0);
+                    double itemHeight = sampleItem.ActualHeight + sampleItem.Margin.Top + sampleItem.Margin.Bottom;
+
+                    // Find index based on dividing number of items by height of each item
+                    int index = Math.Min(DragDropListView2.Items.Count - 1, (int)(pos.Y / itemHeight));
+
+                    // Find the item that we want to drop
+                    ListViewItem targetItem = (ListViewItem)DragDropListView2.ContainerFromIndex(index); ;
+
+                    // Figure out if to insert above or below
+                    Windows.Foundation.Point positionInItem = e.GetPosition(targetItem);
+                    if (positionInItem.Y > itemHeight / 2)
+                    {
+                        index++;
+                    }
+
+                    // Don't go out of bounds
+                    index = Math.Min(DragDropListView2.Items.Count, index);
+
+                    // Finally, add to target listview
+                    contacts2.Insert(index, temp);
+
+                    // Go through source list and remove the items that are being moved
+                    foreach (Contact contact in DragDropListView.Items)
+                    {
+                        if (contact.FirstName == temp.FirstName && contact.LastName == temp.LastName && contact.Company == temp.Company)
+                        {
+                            contacts1.Remove(contact);
+                            break;
+                        }
+                    }
+                }
+                
+                e.AcceptedOperation = DataPackageOperation.Move;
+                def.Complete();
+            }
+        }
+
+        private async void Source_Drop(object sender, DragEventArgs e)
```

Comment:
Can this method and Target_Drop be partially consolidated? Looks like there's a lot of code copy-pasted between the two methods.

Reply:
Both functions do the same thing but calculate/add into different ListViews. I've tried to consolidate the two but because you cant pass in any functions to the argument other than sender and DragEventArgs, I haven't found a way to dynamically find which LV to drop into in one function.

Reply:
I think all the code used to determine temp and index is the same, isn't it? So maybe you can write a function that returns those variables, called from inside Target_Drop and Source_Drop?

Reply:
Okay - after a lot of messing around I found a way to consolidate the two functions into one. The one function is a bit long and repetitive at two points, but I think its better than having two whole functions that are basically the same. See changes in my upcoming commit.